python/pyiceberg/io/pyarrow.py

```diff
@@ -428,11 +428,11 @@ def visit_not_in(self, term: BoundTerm[pc.Expression], literals: Set[Any]) -> pc
 
     def visit_is_nan(self, term: BoundTerm[Any]) -> pc.Expression:
         ref = pc.field(term.ref().field.name)
-        return ref.is_null(nan_is_null=True) & ref.is_valid()
+        return pc.is_nan(ref)
```

Comment:
This probably shouldn't be in this PR right? Seems like an update with a new version of pyarrow?

Reply:
This is actually to make the CI pass. I've [created a PR](https://github.com/apache/arrow/pull/34184) to allow `ref.is_nan()` as well, but this is not released yet.